File: repasoParcial1/punto4/punto4.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Bubble Sort
void bubbleSort(int arr[], int n) {
    int i, j, temp;
    for (i = 0; i < n - 1; i++) {
        for (j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int main() {
    int n = 20000; // tamaño del arreglo
    int *arr = malloc(n * sizeof(int));
    srand(time(NULL));

    for (int i = 0; i < n; i++) {
        arr[i] = rand() % 100000; // números aleatorios
    }

    clock_t inicio = clock();
    bubbleSort(arr, n);
    clock_t fin = clock();

    double tiempo = (double)(fin - inicio) / CLOCKS_PER_SEC;
    printf("Ordenamiento en C con %d elementos: %f segundos\n", n, tiempo);

    free(arr);
    return 0;
}

File: repasoParcial1/punto4/punto4.py
import random
import time

def bubble_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

n = 20000
arr = [random.randint(0, 100000) for _ in range(n)]

inicio = time.time()
bubble_sort(arr)
fin = time.time()

print(f"Ordenamiento en Python con {n} elementos: {fin - inicio} segundos")

